docs: add getting started section (#6737)
meltano/meltano: docs/src/_getting-started/meltano-at-a-glance.md

---
title: Meltano at a Glance
description: If you want to know why you should use Meltano, here's a quick overview of the basics of all of it.
layout: doc
weight: 1
---

> This document is intended to give you enough technical understanding of Meltano to becoming excited about it and wanting to use it! It's not going to teach you how to use it, we got Tutorials & How Tos for that. When you're ready to start your first Meltano project, we recommend you dive right into our Tutorial.

Reviewer: ```suggestion
This document is intended to give you enough technical understanding of Meltano to become excited about it and wanting to use it! It's not going to teach you how to use it, we've got Tutorials & How To's for that. When you're ready to start your first Meltano project, we recommend you dive right into our Tutorial.
```

### What users say

_"For us it's a better day at work when we can use Meltano."_ - Nino Müller, Head of Technology at Substring

> _"I love Meltano because it’s so pleasant to use with its DevOps and Everything-as-Code style. It is easy to set up, flexible, and integrates with pretty much any orchestrator as well as the data built tool (dbt)"._  - Martin Morset

Reviewer: ```suggestion
_"I love Meltano because it’s so pleasant to use with its DevOps and Everything-as-Code style. It is easy to set up, flexible, and integrates with pretty much any orchestrator as well as dbt (data build tool)"._  - Martin Morset
```

## Meltano at a glance

> Welcome to your Open Source DataOps Infrastructure to manage and centralize all the data tools in your stack and turn them into your ideal end-to-end data platform. With Meltano, you'll be able to collaboratively build and improve your data platform like a software project; spinning up a service or tool (Airflow, dbt, Great Expectations, Snowflake, etc) and easily configuring, deploying and managing it through a single control plane.

Reviewer: ```suggestion
Welcome to your Open Source DataOps Infrastructure! With Meltano you can move your data with 10x the developer experience while also managing all of the data tools in your stack. With Meltano, you can collaboratively build and improve your ideal data platform like a software project; spinning up a service or tool (Singer connectors, Airflow, dbt, Great Expectations, Snowflake, etc) and easily configure, deploy, and manage it through a single control plane.
```

I'd love to see this more focused on the data replication piece while still highlighting the entire platform aspect of it.

Author: Sure I'm all up for it. I'm still looking for the place where I can find the latest Marketing plug description, and am randomly copying the ones I find look the most recent.

Author: But, one general note: Are you tied to the use of the word "Singer" inside this context? Because the fact is, this is supposed to sell people, and using "Singer" will always be a bit confusing (you know, they might and likely don't know what Singer is). I would simply use "data connectors", and put another point of "built on the de-facto protocol for data movement: Singer" somewhere.

### Why companies love to build with Meltano

- **No lock-in**: It's open source and has a strong community, you'll always be free.. (derisk)

> - **It's extensible from day 1**: It's super easy to add a custom connection using the SDKs/EDKs.

Reviewer: How much do you want to link out to other properties from here? (thinking EDK / SDK links)

Author: My feeling tells me, not at all. "At a Glance" is supposed to sell people on Meltano, explain the simplicity, scope and thus tell a potential user why this is of value to him.

 So I don't really want them to go anywhere else if it doesn't help to achieve this goal-

> - **Amazing developer experience**: Developers go from start to finish on new data projects, including extraction, loading, transforming & orchestrating data within days.

Reviewer: ```suggestion
- **Amazing developer experience**: Developers go from start to finish on new data projects, including extraction, loading, transforming, & orchestrating data within days.
```

Oxford commas. I will go to the mat for this 😛

Author: Even digital (arguably unnecessary) commas produce a carbon footprint. Good that we dropped sustainability as a value ;)

- **Small surface area**: Features like "inline transformations" remove unnecessary steps from your data pipelines, and help companies stay compliant to security & GDPR regulations.

### Key Features of Meltano, developers will love

> - **Start simple**: Meltano is pip-installable and comes in a prepackaged docker container, you can have your first ELT pipeline running litterally within minutes.

Reviewer: ```suggestion
- **Start simple**: Meltano is pip-installable and comes in a prepackaged docker container, you can have your first ELT pipeline running within minutes.
```

- **DataOps out-of-the-box**: Meltano provides tools that make DataOps best practices easy to use in every project.
- **Integrates with everything**: 300+ natively supported data sources & targets, as well as additional plugins like great expectations or dbt are natively available. 

> - **Easily customizable**: Meltano isn't just extensible, it's built to be extended! SDK & EDK are easy to use, plugins created by others are available on the Hub.

Reviewer: ```suggestion
- **Easily customizable**: Meltano isn't just extensible, it's built to be extended! The SDK for Singer Connectors & EDK for Meltano Components are easy to use. Meltano Hub helps you find all of the connectors and components created across the data community.
```

We haven't had to use EDK in a sentence yet, nor has it been near SDK.

Author: Mh ok. I really want to drop the "Singer" in front of connector though, and write "EDK for other Meltano plugins". We really need to focus in on the language here as it only gets more confusing.

- **Mature system**: Developed since [2018](https://handbook.meltano.com/timeline), runs in production at large companies like GitLab, and currently powers over a million pipeline runs monthly.

> - **First class ELT tooling built-in**: Extract data from any data source, load into any target, use inline maps to transform on data on the fly, and test the incoming data with great expectations, all in one package. 

Reviewer: ```suggestion
- **First class ELT tooling built-in**: Extract data from any data source, load into any target, use inline maps to transform on data on the fly, and test the incoming data, all in one package. 
```

### The Quick Introduction
Waiting to see how Meltano works within 90 secs? We got you covered:

> <iframe width="800" height="632" src="https://www.youtube.com/embed/53WC4kTwbGU" title="From 0 to ELT in 90 seconds with Meltano, tap-gitlab, and target-postgres" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

Reviewer: Love the inline videos. I do wish each of these videos had better title cards. 

The only downside to this video is that it is _old_. A point for later would be to re-run since so much has improved about the command line experience.

Author: Yes I agree, but I think that's still WIP.

## Core Workflow
Meltano helps you to create your end-to-end data stack within minutes.  The core workflow depends on your data stack, but it will usually involve:
1. **Extracting data** from data sources & loading them into targets.
2. **Transforming data** inside a database.
3. **Orchestrating** the extract/load/transform process.

> 4. Adding **additional steps** to the process like testing the data inside transformations, using great expectations, running analyses inside Jupyter notebooks, etc.

Reviewer: ```suggestion
4. Adding **additional steps** to the process like testing the data inside transformations with dbt tests, using Great Expectations, running analyses inside Jupyter notebooks, visualizing data with Superset etc.
```

</div>
### Transforming data
Here's a complete walk-through extending the extract & load to include more CSVs and running a dbt-project over them to transform the data.

<iframe width="800" height="632" src="https://www.youtube.com/embed/pMZmBMeGe3U" title="How to Use Meltano in 5 Minutes" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

### Orchestrating workflows
Meltano uses Airflow as orchestrator for the pipelines. It's as simple as adding Airflow as a plugin to your project and then running

```
meltano schedule add gitlab-to-postgres --extractor tap-gitlab --loader target-postgres --interval @daily
```

to add the schedule. Meltano also provides commands to start an Airflow instance to execute on these schedules. You can find out more about it in the [Orchestrate Data Section](https://docs.meltano.com/guide/orchestration).

### Adding Additional Steps

> Need to add additional steps to your data pipeline? Here's a complete setup also pulling in superset as visualization tool.

Reviewer: ```suggestion
Need to add additional steps to your data pipeline? Here's a complete setup also pulling in Superset as visualization tool.
```
